src/server/pkg/hashtree/cache.go

```diff
@@ -29,3 +32,57 @@ func NewCache(size int) (*Cache, error) {
 	}
 	return &Cache{c}, nil
 }
+
+// MergeCache is an unbounded hashtree cache that can merge the hashtrees in the cache.
+type MergeCache struct {
+	*localcache.Cache
+}
+
+// NewMergeCache creates a new cache.
+func NewMergeCache(root string) *MergeCache {
+	return &MergeCache{
+		Cache: localcache.NewCache(root),
+	}
+}
+
+// Put puts an id/hashtree pair in the cache and reads the hashtree from the passed in io.Reader.
+func (c *MergeCache) Put(id int64, tree io.Reader) (retErr error) {
+	return c.Cache.Put(fmt.Sprint(id), tree)
+}
+
+// Get does a filtered write of id's hashtree to the passed in io.Writer.
+func (c *MergeCache) Get(id int64, w io.Writer, filter Filter) (retErr error) {
+	r, err := c.Cache.Get(fmt.Sprint(id))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Close(); err != nil && retErr == nil {
+			retErr = err
+		}
+	}()
+	return NewWriter(w).Copy(NewReader(r, filter))
+}
```

Comment:
Could you implement the `Delete` method too? Right now the underling `localcache.Cache.Delete` method is exposed but takes a string while `Put` and `Get` take `int64`s.

Reply:
Sure.